src/sentryLogger/sentryLog.ts

```diff
@@ -0,0 +1,54 @@
+import { addBreadcrumb, SeverityLevel } from "@sentry/react";
+import { createCoreLogger, Level } from '@openstax/ts-utils/services/logger';
+import { JsonCompatibleArray, JsonCompatibleStruct, JsonCompatibleValue } from "@openstax/ts-utils/routing";
+
+/**
+ * Creates a logger that creates breadcrumbs using Sentry.
+ * 
+ * More info: https://develop.sentry.dev/sdk/data-model/event-payloads/breadcrumbs/
+ */
+
+const checkTypeOf = (
+  value: JsonCompatibleStruct | JsonCompatibleValue | JsonCompatibleArray,
+  typeValue: string,
+  defaultValue: any,
+) =>
+  value && typeof value === typeValue ? value : defaultValue
+  ;
+
+const flattenObj = (obj: { [x: string]: any; }) =>
```

Comment:
when copying red-code from stack overflow i like to put a link to the SO answer in a comment above the function